### PR TITLE
Feature/bump web3 ipfs

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

Bumps web3 and ipfs to latest versions. Fixes the contract deploy method by passing on kwargs.

## Fixes

- Partially addresses https://github.com/fetchai/agents-aea/issues/1459

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: aea/crypto/ethereum.py [hunk 1]
```diff
@@ -474,9 +474,10 @@ def get_deploy_transaction(  # pylint: disable=arguments-differ
         :returns tx: the transaction dictionary.
         """
         # create the transaction dict
-        nonce = self.api.eth.getTransactionCount(deployer_address)
+        _deployer_address = self.api.toChecksumAddress(deployer_address)
+        nonce = self.api.eth.getTransactionCount(_deployer_address)
         instance = self.get_contract_instance(contract_interface)
-        data = instance.constructor().__dict__.get("data_in_transaction")
+        data = instance.constructor(**kwargs).buildTransaction().get("data", "0x")
```
Comment (DavidMinarsch) [inline aea/crypto/ethereum.py:480]: now kwargs are used!

File: scripts/generate_ipfs_hashes.py [hunk 2]
```diff
@@ -160,7 +160,12 @@ def ipfs_hashing(
     #      use ignore patterns somehow
     # ignore_patterns = configuration.fingerprint_ignore_patterns # noqa: E800
     assert configuration.directory is not None
-    result_list = client.add(configuration.directory, recursive=True)
+    result_list = client.add(
+        configuration.directory,
+        recursive=True,
+        period_special=False,
+        follow_symlinks=False,
+    )
```
Comment (DavidMinarsch) [inline scripts/generate_ipfs_hashes.py:168]: As per discussion here: https://github.com/ipfs-shipyard/py-ipfs-http-client/blob/master/CHANGELOG.md#py-ipfs-http-client-060-30062020

Comment (marcofavorito) [reply scripts/generate_ipfs_hashes.py:168]: why `period_special=False`?

**Review by marcofavorito — APPROVED** [review]
LGTM

**Pull request merged**